src/contracts/interfaces/IPaymentCoordinator.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./IStrategy.sol";
+
+/**
+ * @title Interface for the `IPaymentCoordinator` contract.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @notice Allows AVSs to make "Range Payments", which get distributed amongst the AVSs' confirmed
+ * Operators and the Stakers delegated to those Operators.
+ * Calculations are performed based on the completed Range Payments, with the results posted in
+ * a Merkle root against which Stakers & Operators can make claims.
+ */
+interface IPaymentCoordinator {
+
+    /// STRUCTS ///
+    struct StrategyAndMultiplier {
+        IStrategy strategy;
+        // weight used to compare shares in multiple strategies against one another
+        uint96 multiplier;
+    }
+    struct RangePayment {
+        // Strategies & relative weights of shares in the strategies
+        StrategyAndMultiplier[] strategiesAndMultlipliers;
+        IERC20 token;
```

Comment:
Could probably be good to have a from address here for the transfer so AVSs aren't forced into putting all their payments into the ServiceManager first.

Reply:
funny you should mention that.
an earlier version actually had an 'avs' input parameter, which was removed, but would serve the purpose of not having to route through the ServiceManager.
with the way token approvals function, I don't think there's a safe way to take a 'from' parameter here, unless we have very different ideas on how this would work.